kafka/src/main/java/io/micronaut/configuration/kafka/processor/KafkaConsumerProcessor.java

```diff
@@ -531,34 +531,40 @@ public void process(BeanDefinition<?> beanDefinition, ExecutableMethod<?, ?> met
                                                 }
                                             } catch (Throwable e) {
                                                 handleException(kafkaConsumer, consumerBean, consumerRecord, e);
-                                                continue;
+                                                // break out of the poll loop so that re-delivery can be attempted
+                                                failed = true;
+                                                break;
                                             }
 
-                                            if (offsetStrategy == OffsetStrategy.SYNC_PER_RECORD) {
-                                                try {
-                                                    kafkaConsumer.commitSync(
-                                                            currentOffsets
-                                                    );
-                                                } catch (CommitFailedException e) {
-                                                    handleException(kafkaConsumer, consumerBean, consumerRecord, e);
+                                            if (!failed) {
```

Comment:
`if (!failed)` not necessary here as we break out of the loop. But I recognize it has its value for better readability.